tests/functionality_tests/mission_test.py

```diff
@@ -0,0 +1,56 @@
+# THIS IS A TESTING SCRIPT. USE THIS SCRIPT WHEN PHYSICALLY TESTING THE ROBOT AND CHANGE THE run() FUNCTION BY ADDING THE MISSION STATE YOU WANT TO TEST
+
+from engine.robot import Robot
+from engine.base_station import BaseStation
+from engine.mission import Mission
+from engine.mission import ControlMode
+from engine.database import DataBase
+from engine.robot_state import Robot_State
+from engine.phase import Phase
+
+import math
+
+
+class MissionTest():
+
+    def __init__(self):
+        self.rb_state = Robot_State(0, 0, 0, .2, .5, .2)
+        self.rob = Robot(self.rb_state)
+        self.rob_base = BaseStation(position=(
+            42.444250, -76.483682), heading=math.pi/2, battery=None, plastic_load=None)
+        self.database = DataBase(self.rob)
+        self.m = Mission(robot=self.rob, base_station=self.rob_base,
+                         init_control_mode=ControlMode.LAWNMOWER)
+
+    def test_setup(self):
+        self.rob.execute_setup(self.m.robot_radio_session,
+                               self.m.gps, self.m.imu, self.m.motor_controller)
+        self.rb_state.track_obstacle_thread.stop()
```

Comment:
why do we want to stop the thread for testing scripts? Don't we still need this for monitoring obstacles in our script? Or is it because we are testing the phases in isolation meaning we don't have to keep track of them?

Reply:
yes. we are trying to test the other phases in isolation

Reply:
ok makes sense!